### PR TITLE
Rnn language translation

### LINKED ISSUE #27: Add NLP/machine translation RNN benchmark task

### PR DESCRIPTION
**Pull request by ehoelzl**
Adds the base image for GNMT training on WMT14 dataset.

Updates the base pytorch image to CUDA10, and the image is now in `mlbench/mlbench-pytorch-base:cuda-v10.0`

Some issues were encountered with MPI. This implementation uses mixed precision training with `apex.amp` from [NVIDIA](https://github.com/NVIDIA/apex/tree/master/apex/amp), but `dist.all_reduce` does not work with `float16` tensors. Hence, for now, the tensors are cast into `float32`. 

This issue should not occur when using `NCCL` or `GLOO` (not tested yet), and make training faster in a distributed setting (`NCCL` was originally used in NVIDA's  [implementation](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Translation/GNMT))

### CODE REVIEW

**Comment by mmilenkoski** [comment]
Do we need to update the Dockerfiles in the other images to extend from the new base image with CUDA 10? Are there going to be some compatibility issues?

**Comment by martinjaggi** [comment]
we can do the tests maybe together with updating to pytorch 1.4 ? see #34

**Comment by ehoelzl** [comment]
@mmilenkoski I don't think there should be any problems with CUDA 10, but best to check. Also, we could do that together with updating pytorch. I'm currently building a base image for CUDA 10 and pytorch 1.4.0. Will run the tests to see if anything needs to be changed

File: pytorch/translation/openmpi-wmt14-gnmt-all-reduce/main.py [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+    )
+
+    # Build loss function
+    loss_function = build_criterion(config.PAD, smoothing)
+
+    # Bilingual Evaluation Understudy Score
+    metrics = [BLEUScore()]
+
+    # Partition data
+    train_set = partition_dataset_by_rank(train_set, rank, world_size)
```
Comment (Panaetius) [inline pytorch/translation/openmpi-wmt14-gnmt-all-reduce/main.py]: If you look e.g. here https://github.com/mlbench/mlbench-benchmarks/blob/develop/pytorch/imagerecognition/openmpi-cifar10-resnet20-all-reduce/main.py#L84
we also partition the validation set to the different workers, so each worker has to only do validation on 1/#workers part of the validation set. 
Then `validation_round` aggregates all the validations here https://github.com/mlbench/mlbench-core/blob/2f52d54de9608549087ae7ae6760544d3c554d4d/mlbench_core/controlflow/pytorch/controlflow.py#L197 
This speeds up validation a bit (and after allreduce in the train loop, all workers are in the same state). Minor optimization but still cuts down on total time to a result a bit.

**Review by Panaetius — APPROVED** [review]
Thanks!
